src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs

```diff
@@ -216,13 +212,6 @@ private static void ProcessRequest(HttpWebRequest request)
                 return;
             }
 
-            if (IsRequestInstrumented(request))
```

Comment:
So in the event of automatic retry (due to ssl things, or redirect, or whatever) the same `HttpWebRequest` is reused. Below we patch the state we need onto the `HttpWebRequest` by taking over the the `IAsyncResult` used by the framework. If you remove this check, on retries, we will patch over what we already patched. This makes me nervous. Did you verify everything is cleaned up and we aren't leaking anything? Is the first activity correctly stopped before `StartActivity` is called again?

Reply:
I think to unblock progress, @denisivan0v could you remote the HttpClient (.NETFramework) from this PR? Overall this is good, but since we want to address .NETFW HttpClient concerns, it'd be good to split that our to a separate PR.

Reply:
Sure, I can do it. But would it be okay the instrumentation works differently for .NET/.NET Core and .NETFW HttpClient?

Reply:
@CodeBlanch I've made a quick experiment and it seems the instrumentation doesn't modify `HttpRequestMessage` instance passed to `HttpClient` at all (see https://github.com/open-telemetry/opentelemetry-dotnet/runs/6697287031?check_suite_focus=true#step:5:506). 
Also, the way I understand how `HttpWebRequest` works, it is not really possible to reuse the same `HttpWebRequest` object since these are created per request (and being put to `Connection.m_WriteList`) and reuse the underlying [Connection](https://referencesource.microsoft.com/#System/net/System/Net/_Connection.cs,831) object. 

That said, I don't think removing this check we will patch a previously patched `HttpWebRequest` object we intercept here. Does this make sense?

Reply:
Also, verified that in case automatic redirects (`HttpClientHandler.AllowAutoRedirect = true`) those will be handled by `HttpWebRequest`, so we will be re-instrumenting it and starting new activities. Once requests processed, responses will be handled correctly (`LazyAsyncResult.m_AsyncCallback` will be called), so all corresponding activities will be called. It seems to me everything will be properly cleaned up.

![image](https://user-images.githubusercontent.com/5635171/171533063-b115f916-f3c7-43d4-a439-716d37b9b4b1.png)

Reply:
@CodeBlanch, @cijothomas does the information above clarifies the initial concern or you still want me move this part to a separate PR?

Reply:
Here is an example of HttpWebRequest re-submitting itself: https://referencesource.microsoft.com/#System/net/System/Net/HttpWebRequest.cs,3885

It is a thing! 😄 [TestSecureTransportRetryFailureRequest](https://github.com/open-telemetry/opentelemetry-dotnet/blob/b680ec624c1ad31ebcf4ec0bc45b94679e4e0a4a/test/OpenTelemetry.Instrumentation.Http.Tests/HttpWebRequestActivitySourceTests.netfx.cs#L635) is the test I added to verify one of the cases that I could figure out how to reproduce on demand. It has been a while since I read that reference code, but there are automatic re-submit cases beyond redirects. Most of them are SSL. Upgrades, downgrades, or during negotiation IIRC.

Anyway sorry I didn't realize I was blocking this. Let me take the code on the PR for a spin tomorrow just to sanity check it and then I'll give feedback or approve.

Reply:
OK I took the HttpWebRequest stuff for a spin.

If we remove this check...

https://github.com/open-telemetry/opentelemetry-dotnet/blob/a58c7a3f62dc1e0e308792db3d7eef17faed9e63/src/OpenTelemetry.Instrumentation.Http/Implementation/HttpWebRequestActivitySource.netfx.cs#L219-L224

...what happens is...

* Activity1 is started. Request is hooked. Original state is wrapped with our instrumentation state.
* Some internal exception is thrown. A resubmit is triggered.
* Activity2 is started. Request is hooked again. Our instrumentation state wraps itself.
* Request finishes. Could be successful, could be another exception.
* Response is processed.
* Activity2 is stopped and tagged with the response. Wrapped state is called (this is our first instrumentation state).
* Activity1 is stopped and tagged with the same response as Activity2. Wrapped state is called (the original internal state).
* Original state fires to finish the Request.

This _appears_ to work. But there are issues. The trace looks like this:

![image](https://user-images.githubusercontent.com/28367120/174136132-84a5c817-3eb8-4e56-b250-f13788ae8a18.png)

Activity2 is a child of Activity1. I don't think that is what the spec is describing. Instead of...

```
Root-------------------------
   POST      --------------
        POST          ---
```

It should be...

```
Root--------------
  POST ---
  POST       ---
```

No?

Now we could stop Activity1 before we start Activity2. But there is a problem with that. We don't have a response or an exception to properly clean up Activity1. There is nothing triggered by HttpWebRequest when one of these internal resubmits happens we could listen to.

Basically we only ever get the final response. So if Activity1 failed and Activity2 succeeded, both would end up being decorated as if both succeeded. It would just look like a dupe to end users. Really what we would want to see on Activity1 is the original exception that triggered the resubmit. Then Activity2 we would want to see the final result. But I don't think that is possible looking at the reference source.

I'm thinking... we might be best just leaving this alone ☹️

Reply:
Rolled back HttpWebRequest instrumentation from this PR. Saved changes in https://github.com/denisivan0v/opentelemetry-dotnet/tree/denisivan0v/http-retries-redirects-httpwebrequest branch in case we still need apply them in the future.